### PR TITLE
Create_resource for certonly

### PR DESCRIPTION
**Pull request by lvicainne**
Hello,
I propose to add create_resource in order to be better compliant with Hiera,
Thanks.
Regards

### CODE REVIEW

**Comment by bastelfreak** [comment]
Hi @lvicainne thanks for the PR!

---

@igalic do you have any objections here on using create_resources? AFAIK you're not a big fan of it. Any ideas for a better implementation?

File: manifests/init.pp [hunk 1]
```diff
@@ -103,4 +106,8 @@
     environment => concat([ "VENV_PATH=${venv_path}" ], $environment),
     refreshonly => true,
   }
+
+  $real_certonly = hiera_hash('letsencrypt::certonly', $certonly)
```
Comment (baurmatt) [inline manifests/init.pp]: Please don't do Hiera lookup with the module

File: manifests/init.pp [hunk 2]
```diff
@@ -103,4 +106,8 @@
...
+  create_resources(::letsencrypt::certonly, $real_certonly)
```
Comment (baurmatt) [inline manifests/init.pp]: Please use `each` as it's Puppet 4 native.

[the remark this answers is not in the document]
Comment (bastelfreak) [reply manifests/init.pp]: All of our modules are only tested on Puppet 4.10 an newer: https://github.com/voxpupuli/puppet-letsencrypt/blob/master/metadata.json#L58

**Comment by ody** [comment]
Good day @lvicainne. We certainly appreciate your contribution!

Doing house cleaning and due to inactivity I am going to close this PR. If you are still interested in the proposed functionality then please re-open with previous comments addressed.

Thanks again!

**Pull request closed without merging**